dpnp/dpnp_iface_histograms.py

```diff
@@ -306,6 +312,61 @@ def digitize(x, bins, right=False):
     return bins.size - dpnp.searchsorted(bins[::-1], x, side=side)
 
 
+def _find_supported_dtype(dt, supported):
+    if dt in supported:
+        return dt
+
+    for t in supported:
+        if dpnp.can_cast(dt, t):
+            return t
+
+    return None
+
+
+def _result_type_for_device(dtype1, dtype2, device):
+    rt = dpnp.result_type(dtype1, dtype2)
+    return map_dtype_to_device(rt, device)
+
+
+def _align_dtypes(a_dtype, bins_dtype, ntype, device):
+    has_fp64 = device.has_aspect_fp64
+    a_bin_dtype = _result_type_for_device(a_dtype, bins_dtype, device)
+
+    supported_types = (dpnp.float32, dpnp.int64, numpy.uint64, dpnp.complex64)
+    if has_fp64:
+        supported_types += (dpnp.float64, dpnp.complex128)
+
+    float_types = [dpnp.float32, dpnp.float64]
+    complex_types = [dpnp.complex64, dpnp.complex128]
+
+    a_bin_dtype = _find_supported_dtype(a_bin_dtype, supported_types)
+    hist_dtype = _find_supported_dtype(ntype, supported_types)
+    if hist_dtype == numpy.uint64:
+        hist_dtype = dpnp.int64
+
+    if (a_bin_dtype in float_types and hist_dtype in float_types) or (
+        a_bin_dtype in complex_types and hist_dtype in complex_types
+    ):
+        common_type = _result_type_for_device(a_bin_dtype, hist_dtype, device)
+        a_bin_dtype = common_type
+        hist_dtype = common_type
+
+    if (a_bin_dtype in float_types and hist_dtype in complex_types) or (
+        a_bin_dtype in complex_types and hist_dtype in float_types
+    ):
+        if a_bin_dtype == dpnp.complex128:
+            hist_dtype = dpnp.float64
+        elif a_bin_dtype == dpnp.float64:
+            hist_dtype = dpnp.complex128
+
+        if hist_dtype == dpnp.complex128:
+            a_bin_dtype = dpnp.float64
+        elif hist_dtype == dpnp.float64:
+            a_bin_dtype = dpnp.complex128
```

Comment:
Isn't that do the same like in above block 357-360?

Reply:
No. On lines 357-360 we are updating `hist_dtype` and here we are updating `a_bin_dtype`.
But anyway it is removed